cmd/gcp-controller-manager/node_csr_approver.go

```diff
@@ -601,45 +621,85 @@ func parsePEMBlocks(raw []byte) (map[string]*pem.Block, error) {
 	return blocks, nil
 }
 
-func clusterHasInstance(ctx *controllerContext, instanceZone string, instanceID uint64) (bool, error) {
+// getClusterInstanceGroupUrls returns a list of instance groups for all node pools in the cluster.
+func getClusterInstanceGroupUrls(ctx *controllerContext, instanceZone string) (instanceGroupUrls []string, err error) {
```

Comment:
nit: is there a strong need to have these named?

Reply:
no, removed it.